features/bootstrap/FeatureContext.php

```diff
@@ -0,0 +1,254 @@
+<?php
+
+use Behat\Behat\Context\Context;
+
+use Behat\MinkExtension\Context\RawMinkContext;
+use Behat\Behat\Context\SnippetAcceptingContext;
+use Behat\Behat\Hook\Scope\BeforeScenarioScope;
+use SensioLabs\Behat\PageObjectExtension\PageObject\Page;
+
+/**
+ * Defines application features from the specific context.
+ */
+class FeatureContext extends RawMinkContext implements Context, SnippetAcceptingContext
+{
+    
+    protected $loginPage;
+    protected $title;
+    protected $unreadPage;
+    protected $apiClientPage;
+    protected $toolBar;
+    protected $username;
+    protected $password;
+    
+    public function __construct(
+        LoginPage $loginpage,Title $title,UnreadEntriesPage $unreadPage,APIClientsPage $apiClientPage,ToolBar $toolBar,
+        $parameters){
+            $this->loginPage = $loginpage;
+            $this->title = $title;
+            $this->unreadPage = $unreadPage;
+            $this->apiClientPage= $apiClientPage;
+            $this->toolBar = $toolBar;
+            $this->username=$parameters['adminUserName'];
+            $this->password=$parameters['adminPassword'];
+    }
+    
+    /**
+     * Add a new entry with given url parameter 
+     * @When the user adds a new entry with the url :link
+     * @Given the user has added a new entry with the url :link
+     * @param string $link
+     * 
+     * @return void
+     */
+    public function addEntry($link)
+    {
+        $this->unreadPage->addNewEntry($this->getSession(),$link);
+    }
+    
+    /**
+     * Expect added entry with correct title and link description as given in feature file
+     * @Then an entry should be listed in the list with the title :title and the link description :description
+     * @param string $description
+     * @param string $title
+     * 
+     * @return void
+     */
+    public function entryShouldBeListed($title, $description)
+    {
+        $isEntryListed = $this->unreadPage->isEntryListed($this->getSession(),$title,$description);
+        expect($isEntryListed)->toBe(true);
+    }
+    
+    /**
+     * Expect correct number of unread entries as expected in feature file
+     * @Then the count of unread entries should be :num
+     * @Given the list of unread entries is :num
+     * 
+     * @param integer $num
+     * 
+     * @return void
+     */
+    public function theCountOfUnreadEntries($num)
+    {
+        $unread = $this->unreadPage->countUnreadEntry($this->getSession());
+        expect($unread)->toBe($num);
+    }
+    
+    /**
+     * Open the login page.
+     * @Given the user has browsed to the login page
+     * 
+     * @return Page
```

Comment:
```suggestion
     * @return void
```
As this function doesn't return anything.